### PR TITLE
remotecommand: close resize channel for notification

### PR DESCRIPTION
**Pull request by fuweid**
Remotecommand package should notify executor by closing resizeChan.

Signed-off-by: Wei Fu <fuweid89@gmail.com>

**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Remotecommand package should notify executor by closing resizeChan.

**Which issue(s) this PR fixes**:

no issue track in kubernetes repo but in containerd
https://github.com/containerd/cri/pull/1418

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by matthyx** [comment]
```release-note
NONE
```

**Comment by fuweid** [comment]
@matthyx thanks! updated it.

**Comment by matthyx** [comment]
> @matthyx thanks! updated it.

can you put `NONE` inside?

**Comment by fuweid** [comment]
> > @matthyx thanks! updated it.
> 
> can you put `NONE` inside?

done.

**Comment by fuweid** [comment]
@mattjmcnaughton no idea about failed cases. could you help to retest it? thanks!

**Comment by mattjmcnaughton** [comment]
/retest

@fuweid its possible the test failures are just flake. For the future, after its marked `/ok-to-test`, I believe that you should have the permissions to issue the `/retest` command and trigger a retesting of all failed tests :)

**Comment by matthyx** [comment]
Maybe this is not a flake after all?

**Comment by fuweid** [comment]
@matthyx not sure and is there any way to check containerd log in ci?

**Comment by matthyx** [comment]
I dunno :( maybe @liggitt could help?

File: pkg/kubelet/server/remotecommand/httpstream.go [hunk 1]
```diff
@@ -116,7 +116,10 @@ func createStreams(req *http.Request, w http.ResponseWriter, opts *Options, supp
 
 	if ctx.resizeStream != nil {
 		ctx.resizeChan = make(chan remotecommand.TerminalSize)
-		go handleResizeEvents(ctx.resizeStream, ctx.resizeChan)
+		go func() {
+			handleResizeEvents(ctx.resizeStream, ctx.resizeChan)
+			close(ctx.resizeChan)
```
Comment (liggitt) [inline pkg/kubelet/server/remotecommand/httpstream.go:121]: would it make sense to move this into `handleResizeEvents` as a `defer close(channel)`?

Comment (fuweid) [reply pkg/kubelet/server/remotecommand/httpstream.go]: I was thinking that it is easy to check if we make `close(ctx.resizeChan)` closed to `make(chan remotecommand.TerminalSize)` statement. But your suggestion sgtm. updated it.

Comment (derekwaynecarr) [reply pkg/kubelet/server/remotecommand/httpstream.go]: I am fine with it in the present location.  In reviewing, I had initially looked for where the channel was created initally as well, but its fine here.

**Comment by fuweid** [comment]
thanks @matthyx @liggitt @mattjmcnaughton @derekwaynecarr for the review!

**Pull request merged**